fix.go

```diff
@@ -0,0 +1,135 @@
+package kubeaudit
+
+import (
+	"bytes"
+
+	"github.com/Shopify/kubeaudit/internal/k8s"
+	"github.com/Shopify/kubeaudit/internal/yaml"
+	"github.com/Shopify/kubeaudit/k8stypes"
+)
+
+func fix(results []Result) ([]byte, error) {
+	var outputBytes [][]byte
+	var newResources []k8stypes.Resource
+
+	if len(results) == 0 {
+		return []byte{}, nil
+	}
+
+	// Fix all the resources
+	for _, result := range results {
+		for _, auditResult := range result.GetAuditResults() {
+			newResources = append(newResources, auditResult.Fix(result.GetResource().Object())...)
+		}
+	}
+
+	// Convert all the resources to bytes
+	for _, result := range results {
+		if result.GetResource().Object() == nil {
+			outputBytes = append(outputBytes, result.GetResource().Bytes())
+			continue
+		}
+
+		fixedresourceBytes, err := resourceToBytes(result.GetResource().Object(), result.GetResource().Bytes())
+		if err != nil {
+			return nil, err
+		}
+
+		outputBytes = append(outputBytes, fixedresourceBytes)
+	}
+
+	// Convert all the new resources to bytes
+	for _, newResource := range newResources {
+		fixedresourceBytes, err := resourceToBytes(newResource, nil)
+		if err != nil {
+			return nil, err
+		}
+		outputBytes = append(outputBytes, fixedresourceBytes)
+	}
+
+	fixedManifest := bytes.Join(outputBytes, []byte("---"))
+
+	return fixedManifest, nil
+}
+
+func resourceToBytes(fixedResource k8stypes.Resource, origResourceBytes []byte) ([]byte, error) {
+	fixedresourceBytes, err := k8s.EncodeResource(fixedResource)
+	if err != nil {
+		return nil, err
+	}
+
+	if origResourceBytes == nil {
+		// This is a new resource (not in the original manifest)
+		// Add  a leading newline
+		fixedresourceBytes = append([]byte{'\n'}, fixedresourceBytes...)
+	} else {
+		fixedresourceBytes, err = yaml.Merge(origResourceBytes, fixedresourceBytes)
+		if err != nil {
+			return nil, err
+		}
+
+		// Add any leading and trailing whitespace that was present in the original
+		fixedresourceBytes = bytes.Replace(origResourceBytes, bytes.TrimSpace(origResourceBytes), fixedresourceBytes, 1)
+
+		// Remove the redundant trailing newline
+		if fixedresourceBytes[len(fixedresourceBytes)-1] == '\n' {
+			fixedresourceBytes = fixedresourceBytes[:len(fixedresourceBytes)-1]
+		}
+	}
+
+	fixedresourceBytes, err = cleanupManifest(origResourceBytes, fixedresourceBytes)
+	if err != nil {
+		return nil, err
+	}
+
+	return fixedresourceBytes, nil
+}
+
+// TODO do this better??
```

Comment:
I didn't change this code but it seems pretty finicky to me. As in, I think in some cases it produces invalid yaml. What do you think? If it's a reasonable concern and not an easy fix I'll open an issue for it.

Reply:
Oh true! 
How do you know it produces bad YAML? I think we should definitely address that (probably not in the scope of this PR, though)

Reply:
Because it just remove lines of YAML
eg.
```
kind: Pod
metadata:
    creationTimestamp: null
spec:
    containers:
    - name: AAcontainer
```
would turn into
```
kind: Pod
metadata:
spec:
    containers:
    - name: AAcontainer
```
(where `creationTimestamp: null` is removed)
which maybe is valid yaml but I don't think it's a valid kubernetes config. I think the correct fix for this is upstream. I had a quick look and they do have it set to only marshal that field if it's not empty, but because the type is a date object it can't tell when it's empty. There are other cases where they use a pointer to a date object and setting that to null won't produce the field in the yaml, which is what we want. Not sure why some dates are pointers and others aren't...

Reply:
Ahhh I see. I think it's worth creating an issue on our end to keep track of things and explore this farther as well? Thoughts?

Reply:
Done https://github.com/Shopify/kubeaudit/issues/240